command/agent.go

```diff
@@ -332,10 +324,92 @@ func (c *AgentCommand) Run(args []string) int {
 		EnableReauthOnNewCredentials: config.AutoAuth.EnableReauthOnNewCredentials,
 	})
 
-	// Start things running
+	// Start auto-auth and sink servers
 	go ah.Run(ctx, method)
 	go ss.Run(ctx, ah.OutputCh, sinks)
 
+	// Parse agent listener configurations
+	if config.Cache != nil && len(config.Cache.Listeners) != 0 {
+		cacheLogger := c.logger.Named("cache")
+
+		// Create the API proxier
+		apiProxy := cache.NewAPIProxy(&cache.APIProxyConfig{
+			Logger: cacheLogger.Named("apiproxy"),
+		})
+
+		// Create the lease cache proxier and set its underlying proxier to
+		// the API proxier.
+		leaseCache, err := cache.NewLeaseCache(&cache.LeaseCacheConfig{
+			BaseContext: ctx,
+			Proxier:     apiProxy,
+			Logger:      cacheLogger.Named("leasecache"),
+		})
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error creating lease cache: %v", err))
+			return 1
+		}
+
+		// Create a muxer and add paths relevant for the lease cache layer
+		mux := http.NewServeMux()
+		mux.Handle("/v1/agent/cache-clear", leaseCache.HandleCacheClear(ctx))
```

Comment:
I thought the plan was to have different endpoints for clearing different parts of the cache (eventually). Given that, I'd put this behind a reasonable collection, e.g. /v1/agent/cache/clear-all.

I also wonder whether we need a /v1/ at all. If we don't use that for cache paths, we can automatically know that any path coming in starting with /v1/ is destined for Vault.

Reply:
I removed the `/v1` prefix for the API as it makes total sense and the change was easy as well. As to the other point on breaking the API into parts, I'll wait for other thoughts as well. 

On the face of it, it does feel like breaking the API into their own endpoints is a good thing. Specially keeping in mind that agent might continue to grow in its feature set. Managing this single API to handle all the things in future might be problematic.

Reply:
The endpoint is a public API exposed by the agent, albeit a small one, so I think having it versioned to `/v1` is reasonable.

Reply:
The single endpoint supports clearing different parts the cache. You'd provide "request_path",  "token", "lease_id", or "all" as the type along with the value in the request body. We had it as a single endpoint in the RFC, but I'm working on getting it updated to be separate endpoints. I think that makes sense.

Reply:
The problem with /v1/ is that if the agent ever uses a path that we want to use in Vault, now there is a clash. If you *must* version the api, then I'd suggest a pattern like `agent/v1` instead of `v1/agent`. Then there can be no confusion or overlap between a Vault path and an agent path.

Having seen the code for the cleaning endpoint I think having the single endpoint makes sense.

Reply:
Addressed in https://github.com/hashicorp/vault/pull/6257